.circleci/config.yml

```
@@ -59,52 +59,47 @@ commands:
         description: 'Set to true if you intend to any browser (for example with playwright).'
 
     steps:
-      - run:
-          name: View install environment
-          command: |
-            node --version
-            yarn --version
       - restore_cache:
-          name: Restore yarn cache
+          name: Restore pnpm cache
           keys:
-            - v8-yarn-{{ checksum "yarn.lock" }}
-      - run:
-          name: Set yarn cache folder
-          command: |
-            # Keep path in sync with `save_cache` for key "v8-yarn-"
-            yarn config set cache-folder /tmp/yarn-cache
-            # Debug information
-            yarn cache dir
-            yarn cache list
+            - pnpm-packages-{{ checksum "pnpm-lock.yaml" }}
       - when:
           condition: << parameters.browsers >>
           steps:
+            - run:
+                name: Install pnpm package manager
+                command: |
+                  corepack enable
+                  corepack prepare pnpm@latest-8 --activate
             - run:
                 name: Prepare playwright hash
-                command: yarn --json list --pattern playwright > /tmp/playwright_info.json
+                command: pnpm list --json --filter playwright > /tmp/playwright_info.json
             - store_artifacts:
                 name: Debug playwright hash
                 path: /tmp/playwright_info.json
             - restore_cache:
                 name: Restore playwright cache
                 keys:
                   - v6-playwright-{{ arch }}-{{ checksum "/tmp/playwright_info.json" }}
+      - run:
+          name: View install environment
+          command: |
+            node --version
+            pnpm --version
       - run:
           name: Install js dependencies
-          command: yarn install
+          command: pnpm install
       - when:
           condition: << parameters.browsers >>
           steps:
             - run:
                 name: Install playwright browsers
                 command: npx playwright install --with-deps
       - save_cache:
-          name: Save yarn cache
-          key: v8-yarn-{{ checksum "yarn.lock" }}
+          name: Save pnpm package cache
+          key: pnpm-packages-{{ checksum "pnpm-lock.yaml" }}
           paths:
-            # Keep path in sync with "Set yarn cache folder"
-            # Can't use environment variables for `save_cache` paths (tested in https://app.circleci.com/pipelines/github/mui/material-ui/37813/workflows/5b1e207f-ac8b-44e7-9ba4-d0f9a01f5c55/jobs/223370)
-            - /tmp/yarn-cache
+            - node_modules
```

Should cache the pnpm store instead?
```suggestion
            - .pnpm-store
```
This would require running `pnpm config set store-dir .pnpm-store` after installing pnpm – see https://pnpm.io/continuous-integration#circleci

---

Thanks for spotting it. This is a great question. 👍 
On one hand, it seems to make sense, but on the other hand, I think that it won't change anything given our current setup.
For now, our cache is hashed based on the `lock` file.
Whenever anything in the lock file changes, the store will not be used and all new dependencies will be downloaded. 🤔 
This is the same setup we had with `yarn`.

WDYT @mui/code-infra, does it make sense to cache the pnpm store instead, as suggested in the tool documentation? 🤔 
Or should the pnpm cache be removed altogether?
I've checked the latest configuration for `base-ui`, `toolpad`, and `material-ui` and none have explicit caching config, even though it was present in the migration PRs. 🙈 🤷

---

We've removed cache altogether in core and Toolpad, but ymmv 
https://github.com/mui/material-ui/pull/40670 and https://github.com/mui/mui-toolpad/pull/3133

---

> We've removed cache altogether in core and Toolpad, but on X ymmv [mui/material-ui#40670](https://github.com/mui/material-ui/pull/40670) and [mui/mui-toolpad#3133](https://github.com/mui/mui-toolpad/pull/3133)

Hmm, this makes sense to me if we miss the cache often 👍🏻

---

Thanks for the insight regarding the cache.
I've checked our case:
- [Cache hit](https://app.circleci.com/pipelines/github/mui/mui-x/57033/workflows/b8c69212-a4b1-40cd-b8ba-654f1bd2d7a6/jobs/326232) ~50-60s
- [Cache miss](https://app.circleci.com/pipelines/github/mui/mui-x/57028/workflows/d23c2500-c030-4608-9cef-c98063de71d3/jobs/326201) ~60-70s (would be without saving the cache)

Seems like there is indeed no need for it as we basically save no time and use storage, which probably costs more money. 🙈 🤷 
Do you agree that we can drop the `pnpm cache` on x as well @Janpot @cherniavskii?

---

Yep, go for it 👍

---

Possibly a worst-case/slowest [scenario](https://app.circleci.com/pipelines/github/mui/mui-x/57118/workflows/4975d4cd-91de-4969-a457-7d16d702ca4f/jobs/326722): 75s